### DISCUSSION TITLE
Support for RAR Archives / Generalizing archive options

### DISCUSSION THREAD

**Discussion by shakeyourbunny** [MikeMaximus/gbm#234] [Ideas]
Would it be possible to support other compression options than just 7-Zip?

The archive format is still flakey and there have been several occasions when a newer revision of 7-Zip could not open an archive created by an older version of the same application or just produced garbage on decompressing it.

As far as I can tell, GBM just executes the command line 7-Zip in a subprocess, so it would really be better to support other formats. The current dialogue just should have a drop down selection for the used archiver and general compression ratio (default is current 7-Zip profile). 

In addition, a button should be there which leads to a dialog box for creating additional profiles which should contain an entry for the command-line executable, command switches for decompressing, command switches for the compression profiles and additional line for generic switches which apply on all (compression) profiles. A "test" / "find executable" button should be there too, validate the executable.

For Restoring, GBM just looks at the archive extension and uses the corresponding archives automatically.

**Comment by shakeyourbunny** [comment]
The compression profiles should be standardized in order to make the archiver configuration dialog easier to implement (no dynamic entries  / modifications in the archiver configuration dialog).

I think 
- "No Compression": compression level zero, just archiving
- "Standard": standard archiver compression settings
- "High": highest archiver compression level, but no tricks
- "Best": like "High", but with the usual tricks like LZMA

Also, warn if solid archiving is detected. This gives better overall compression, but if the archive is damaged, the whole thing is busted.

**Comment by MikeMaximus** [comment]
Yes supporting other archivers is something I've wanted to do and it's on the list.  

I'm currently working on a side project to clean up GBM and make features much easier to add and improve.  The code has become a bit of a mess over years of adding features as fast as possible.

I'm going to move this to the discussion section for now as it's more of a longer term goal at this point.  Thank you for the detailed suggestions and feel free to add more.
